gcp_airflow_foundations/source_class/source.py

```diff
@@ -54,60 +45,129 @@ def build_dags(self):
         data_source = self.config.source
         logging.info(f"Building DAG for {data_source.name}")
 
+        table_names_to_ingest = [self.config.tables[i].table_name for i in range(len(self.config.tables))]
+        templated_config = False
+        if data_source.regex_matching:
+            full_ingestion_options = data_source.full_ingestion_options
+            if full_ingestion_options.dag_creation_mode == "SOURCE":
+                return self.build_dags_source_level()
```

Comment:
Code would be cleaner if you could do 

  if full_ingestion_options.dag_creation_mode == "SOURCE":
                return self.build_dags_source_level()
 else 
                 return self.build_dags_table_level()